kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,113 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+fn has_asterisk_in_middle(input: &str) -> bool {
+    let input = input.trim(); // Remove leading and trailing whitespace
+    if input.starts_with('*') || input.ends_with('*') {
+        return false;
+    }
+
+    let asterisk_position = input.find('*');
+    if let Some(pos) = asterisk_position {
+        if input[pos - 1..].contains(".*.") {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn to_regex_string_new(glob: &str) -> String {
+    let mut input = String::from(glob);
+
+    //check if asterisk in middle
+    if has_asterisk_in_middle(glob) {
+        input = glob.replace(".*.", ".");
+    }
+    // Construct regular expression
+
+    // Make sure we match the whole thing
+    // Start from the beginning
+    let mut re = String::from("^");
+
+    if input.ends_with(".*") {
+        re.push_str(input.replace(".*", ".[^.]+").as_str());
+    } else if input.ends_with(".**") {
+        re.push_str(input.replace(".**", ".*").as_str());
+    } else {
+        // Replace all "standalone" wildcards with ".*"
+        re.push_str(
+            &input
+                .split('.')
+                .map(|part| match part {
+                    "*" => "[^.]+",
+                    "**" => ".*",
+                    other => other,
+                })
+                .collect::<Vec<&str>>()
+                .join(r"\."),
+        );
+    }
+
+    // If it doesn't already end with a wildcard, add it
+    if !re.ends_with(".*") && !input.contains('*') {
+        re.push_str(".*");
+    }
+
+    // And finally, make sure we match until EOL
+    re.push('$');
+
+    re
+}
+
+pub fn to_regex_new(glob: &str) -> Result<Regex, Error> {
+    let re = to_regex_string_new(glob);
+    Regex::new(&re).map_err(|_err| Error::RegexError)
+}
+
+pub fn matches_path_pattern(input: &str) -> bool {
+    if input.starts_with("*.") {
+        return false;
+    }
+    let pattern = r"^(?:\w+(?:\.\w+)*)(?:\.\*+)?$";
+    let regex = Regex::new(pattern).unwrap();
+
+    let pattern2 = r"^(?:[^.]+\.)*(?:\*{1,2})\.(?:[^.]+\.)*(?:[^.]+)*$";
+    let regex2 = Regex::new(pattern2).unwrap();
+
+    regex.is_match(input) || regex2.is_match(input)
```

Comment:
Ok, so adding validation of paths is obviously a good thing (since this was not done at all before this PR).

The problem is that these two regex:es do not correctly identify whether a path is valid or not.

#### First regex
```rust
let pattern = r"^(?:\w+(?:\.\w+)*)(?:\.\*+)?$";
```
##### Commented
```rust
let pattern = r"(?x)
    # anchor at beginning
    ^
    # one occurrence of
    (?:
        # at least one word character
        \w+
        # followed by zero or more
        (?:       
            \.    # dot
            \w+   # at least one character
        )*
    )
    # followed by zero or one
    (?:
        \.        # dot
        \*+       # at least one asterisk
    )?
    # anchor at end
    $
";
```
This is pretty straight forward, but doesn't cover everything.
It matches paths consisting of word characters only and (optionally) ends with a single `.*`. E.g. `Vehicle.Cabin` or `Vehicle.Cabin.Sunroof.*`

#### Second regex
```rust
let pattern2 = r"^(?:[^.]+\.)*(?:\*{1,2})\.(?:[^.]+\.)*(?:[^.]+)*$";
```
##### Commented
```rust
let pattern = r"(?x)
    # anchor at beginning
    ^
    # zero or more
    (?:
        [^.]+     # at least one non-dot
        \.        # dot
    )*
    # followed by (exactly one)
    (?:
        \*{1,2}   # one or two asterisks
    )
    # followed by (exactly one)
    \.            # dot
    # followed by zero or more
    (?:
        [^.]+     # at least one non-dot
        \.        # dot
    )*
    # followed by any number of
    (?:
        [^.]+     # at least one non-dot
    )*
    #anchor at end
    $
";
```
The second regex is a lot harder to understand and I don't think it's matching the right things (maybe _because_ it's hard to follow the logic). For example, it will allow internal wildcards like `Vehicle.C*.Position` and any number of wildcards like `Vehicle.*.Sunroof.******` or wildcard paths containing invalid characters like `Vehicle.*.#@(&*@#(&#Position`. 

#### Suggested regex
My suggestion is to use something like what is found in `src/jwt/scope.rs` adopted to support `**` as well. Something like this:

##### Commented
```rust 
let re = regex::Regex::new(
    r"(?x)
    ^  # anchor at start (only match full paths)
    
    # At least one subpath (consisting of either of three):
    (?:
        [A-Z][a-zA-Z0-1]*  # An alphanumeric name (first letter capitalized)
        |
        \*                 # An asterisk
        |
        \*\*               # A double asterisk
    )

    # which can be followed by ( separator + another subpath )
    # repeated any number of times
    (?:
        \. # Separator, literal dot
        (?:
            [A-Z][a-zA-Z0-1]*  # Alphanumeric name (first letter capitalized)
            |
            \*                 # An asterisk
            |
            \*\*               # A double asterisk
        )
    )*
    $  # anchor at end (to match only full paths)
    ",
)
.unwrap();
```

or the short version:
```rust
r"^(?:[A-Z][a-zA-Z0-1]*|\*|\*\*)(?:\.(?:[A-Z][a-zA-Z0-1]*|\*|\*\*))*$"
```
(but I really think it's preferable to use the commented version, in the code, as it makes it easier to follow the intended logic, i.e. make the code readable)